src/cli/domain/handle-dependencies/require-template.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const format = require('stringformat');
+const path = require('path');
+
+const cleanRequire = require('../../../utils/clean-require');
+const isTemplateLegacy = require('./is-template-legacy');
+const isTemplateValid = require('../../../utils/is-template-valid');
+const strings = require('../../../resources');
+
+module.exports = function(template, options) {
+  const requireOptions = options || {};
+  let ocTemplate;
+
+  if (isTemplateLegacy(template)) {
+    template = `oc-template-${template}`;
+  }
+
+  if (requireOptions.compiler) {
+    template = `${template}-compiler`;
+  }
+
+  const localTemplate = path.join(__dirname, '../../node_modules', template);
+  const relativeTemplate = path.resolve('.', 'node_modules', template);
+  const componentRelativePath = path.join(
+    requireOptions.componentPath,
+    'node_modules',
+    template
+  );
+
+  [
+    componentRelativePath,
+    template,
+    localTemplate,
+    relativeTemplate
+  ].forEach(pathToTry => {
+    ocTemplate = ocTemplate || cleanRequire(pathToTry, { justTry: true });
```

Comment:
why not using `some` instead of forEach so that you breakout of the loop as soon it get required? Am I missing something?

Reply:
I don't know about some. Perhaps let's do it in another PR as optimisation?